features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/typing/TypingNotificationView.kt

```diff
@@ -0,0 +1,104 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.pluralStringResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import io.element.android.compound.theme.ElementTheme
+import io.element.android.features.messages.impl.R
+import io.element.android.libraries.designsystem.preview.ElementPreview
+import io.element.android.libraries.designsystem.preview.PreviewsDayNight
+import io.element.android.libraries.designsystem.theme.components.Text
+import io.element.android.libraries.matrix.api.room.RoomMember
+import kotlinx.collections.immutable.ImmutableList
+
+@Composable
+fun TypingNotificationView(
+    state: TypingNotificationState,
+    modifier: Modifier = Modifier,
+) {
+    if (state.typingMembers.isEmpty() || !state.renderTypingNotifications) return
+    val typingNotificationText = computeTypingNotificationText(state.typingMembers)
+    Text(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(horizontal = 24.dp, vertical = 2.dp),
+        text = typingNotificationText,
+        overflow = TextOverflow.Ellipsis,
+        maxLines = 1,
+        style = ElementTheme.typography.fontBodySmRegular,
+        color = ElementTheme.colors.textSecondary,
+    )
+}
+
+@Composable
+private fun computeTypingNotificationText(typingMembers: ImmutableList<RoomMember>): AnnotatedString {
+    val names = when (typingMembers.size) {
+        0 -> "" // Cannot happen
+        1 -> typingMembers[0].disambiguatedDisplayName
+        2 -> stringResource(
+            id = R.string.screen_room_typing_two_members,
+            typingMembers[0].disambiguatedDisplayName,
+            typingMembers[1].disambiguatedDisplayName,
+        )
+        else -> pluralStringResource(
+            id = R.plurals.screen_room_typing_many_members,
+            count = typingMembers.size - 2,
+            typingMembers[0].disambiguatedDisplayName,
+            typingMembers[1].disambiguatedDisplayName,
+            typingMembers.size - 2,
+        )
+    }
+    // Get the translated string with a fake pattern
+    val tmpString = pluralStringResource(
+        id = R.plurals.screen_room_typing_notification,
+        count = typingMembers.size,
+        "<>",
+    )
+    // Split the string in 3 parts
+    val parts = tmpString.split("<>")
+    // And rebuild the string with the names
+    return buildAnnotatedString {
+        append(parts[0])
+        withStyle(SpanStyle(fontWeight = FontWeight.Bold)) {
+            append(names)
+        }
+        append(parts[1])
```

Comment:
I wonder if this will work fine for all languages.

Reply:
I have done some check and I think it will be fine. It may causes issue if the string template is not correct (`%1$s` missing for instance). The code could be more robust by checking the size of `parts`.